packages/features/form-builder/FormBuilderField.tsx

```diff
@@ -0,0 +1,337 @@
+          readOnly={readOnly}
+          value={value as string}
+          setValue={setValue as (arg: typeof value) => void}
+        />
+      </WithLabel>
+    );
+  }
+
+  if (componentConfig.propsType === "boolean") {
+    return (
+      <WithLabel field={field} readOnly={readOnly}>
+        <componentConfig.factory
+          name={field.name}
+          label={field.label}
+          readOnly={readOnly}
+          value={value as boolean}
+          setValue={setValue as (arg: typeof value) => void}
+          placeholder={field.placeholder}
+        />
+      </WithLabel>
+    );
+  }
+
+  if (componentConfig.propsType === "textList") {
+    return (
+      <WithLabel field={field} readOnly={readOnly}>
+        <componentConfig.factory
+          placeholder={field.placeholder}
+          name={field.name}
+          label={field.label}
+          readOnly={readOnly}
+          value={value as string[]}
+          setValue={setValue as (arg: typeof value) => void}
+        />
+      </WithLabel>
+    );
+  }
+
+  if (componentConfig.propsType === "select") {
+    if (!field.options) {
+      throw new Error("Field options is not defined");
+    }
+
+    return (
+      <WithLabel field={field} readOnly={readOnly}>
+        <componentConfig.factory
+          readOnly={readOnly}
+          value={value as string}
+          name={field.name}
+          placeholder={field.placeholder}
+          setValue={setValue as (arg: typeof value) => void}
+          options={field.options.map((o) => ({ ...o, title: o.label }))}
+        />
+      </WithLabel>
+    );
+  }
+
+  if (componentConfig.propsType === "multiselect") {
+    if (!field.options) {
+      throw new Error("Field options is not defined");
+    }
+    return (
+      <WithLabel field={field} readOnly={readOnly}>
+        <componentConfig.factory
+          placeholder={field.placeholder}
+          name={field.name}
+          readOnly={readOnly}
+          value={value as string[]}
+          setValue={setValue as (arg: typeof value) => void}
+          options={field.options.map((o) => ({ ...o, title: o.label }))}
+        />
+      </WithLabel>
+    );
+  }
+
+  if (componentConfig.propsType === "objectiveWithInput") {
+    if (!field.options) {
+      throw new Error("Field options is not defined");
+    }
+    if (!field.optionsInputs) {
+      throw new Error("Field optionsInputs is not defined");
+    }
+    return field.options.length ? (
+      <WithLabel field={field} readOnly={readOnly}>
+        <componentConfig.factory
+          placeholder={field.placeholder}
+          readOnly={readOnly}
+          name={field.name}
+          value={value as { value: string; optionValue: string }}
+          setValue={setValue as (arg: typeof value) => void}
+          optionsInputs={field.optionsInputs}
+          options={field.options}
+          required={field.required}
+        />
+      </WithLabel>
+    ) : null;
+  }
+
+  if (componentConfig.propsType === "variants") {
```

Comment:
Separate handling for Component with propsType "variants".
Shouldn't we read propsType from fieldTypeConfigMap?